### PR TITLE
Feature/car crash

### PR DESCRIPTION
**Pull request by DominikGuzei**
This request adds code to handle car crashing:
- adds small vector helper class to calculate angles
- adds `traction` to car controller as threshold for crashing
- introduces second driving behavior for crashing (keep driving into last vector direction)
- removes possibility to accelerate car while crashing

open for Feedback!

### CODE REVIEW

**Comment by stravid** [comment]
Maybe change `new Vector` to `Vector.create` in order to have a consistent codebase.

File: app/assets/javascripts/helpers/math/vector.js.coffee [hunk 1]
```diff
@@ -0,0 +1,29 @@
+
+#= require helpers/namespace
+
+namespace 'helpers.math'
+
+class helpers.math.Vector
+
+  constructor: () ->
```
Comment (roboshoes) [inline app/assets/javascripts/helpers/math/vector.js.coffee]: I really don't think those () are necessary!!

File: app/assets/javascripts/helpers/math/vector.js.coffee [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  constructor: (@x, @y) ->
+
+  length: ->
+    Math.sqrt (@x*@x) + (@y*@y)
+
+  dot: (otherVector) ->
+    @x * otherVector.x + @y * otherVector.y
+
+  # returns angle in radian
```
Comment (DominikGuzei) [inline app/assets/javascripts/helpers/math/vector.js.coffee]: wrong comment -> it returns angle in degrees

File: app/assets/javascripts/helpers/math/vector.js.coffee [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    theta = @dot(otherVector) / (@length() * otherVector.length())
+
+    if theta < -1 then theta = -1
+    if theta > 1 then theta = 1
+
+    Math.acos(theta) * 180 / Math.PI
+
+  # ember like static constructor method
+  # that takes an literal with eiter 'x' and 'y'
+  # or two points 'from' and 'to'
```
Comment (DominikGuzei) [inline app/assets/javascripts/helpers/math/vector.js.coffee]: remove comment - the method is readable enough to grasp the purpose

File: app/assets/javascripts/helpers/math/vector.js.coffee [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+namespace 'helpers.math'
+
+class helpers.math.Vector
+
+  constructor: (@x, @y) ->
+
+  length: ->
+    Math.sqrt (@x*@x) + (@y*@y)
+
+  dot: (otherVector) ->
```
Comment (DominikGuzei) [inline app/assets/javascripts/helpers/math/vector.js.coffee]: could be a one liner

File: app/assets/javascripts/helpers/math/vector.js.coffee [hunk 5]
```diff
@@ -0,0 +1,33 @@
+
+#= require helpers/namespace
+
+namespace 'helpers.math'
+
+class helpers.math.Vector
+
+  constructor: (@x, @y) ->
+
+  length: ->
```
Comment (DominikGuzei) [inline app/assets/javascripts/helpers/math/vector.js.coffee]: could be a one liner

File: app/assets/javascripts/game/controllers/car_controller.js.coffee [hunk 6]
```diff
@@ -1,47 +1,92 @@
...
+    @lengthAtTrack += @speed
+    @_calculatePositionOnPath()
+    @_updateCarPosition()
+    @_checkForFinish()
+
+  _checkForFinish: ->
+    if @lengthAtTrack > @trackLength
+      ($ this).trigger 'crossFinishLine'
+
   _updateTrackLength: ->
```
Comment (DominikGuzei) [inline app/assets/javascripts/game/controllers/car_controller.js.coffee]: could be a one liner

**Comment by DominikGuzei** [comment]
:+1: all changes are done -> please merge!

**Pull request merged**